docs/administration/configuration/configuring-storage-volumes/index.md

```diff
@@ -0,0 +1,194 @@
+
+### NFS
+
+You can deploy an NFS file system inside your cluster using the official [NFS Server Provisioner](https://github.com/helm/charts/tree/master/stable/nfs-server-provisioner) Helm chart.
+
+1. Create your configuration file, `nfs-provisioner-values.yaml`, following the [documentation](https://github.com/helm/charts/tree/master/stable/nfs-server-provisioner#recommended-persistence-configuration-examples) and adapting it to your needs. Remember to name the storage class as REANA expects it, `<helm-release-prefix>-shared-volume-storage-class`.
+
+    ```yaml hl_lines="3"
+    storageClass:
+      defaultClass: true
+      name: reana-shared-volume-storage-class
+    ```
+
+2. Install the NFS Server Provisioner
+
+    ```{ .console .copy-to-clipboard }
+    $ helm install reana-dev-storage stable/nfs-server-provisioner \
+                  -f nfs-provisioner-values.yaml
+    ```
+
+3. Configure the NFS storage volume in your Helm configuration file
+
+    ```{ .yaml .copy-to-clipboard }
+    shared_storaged: # or `infrastructure_storage`
+      backend: nfs
+      volume_size: 200
+      access_modes: ReadWriteMany
+    ```
+
+### HostPath
+
+!!! warning
+    Using HostPath volumes is not recommended, as they present many security risks. See the related [Kubernetes documentation page](https://kubernetes.io/docs/concepts/storage/volumes/#hostpath) for more information.
+
+REANA also supports using directories already present on the host node's filesystem. Note that this approach is not suitable for multi-node deployments, but only for local single-node instances of REANA.
+
+To use a local directory as a storage volume, set the [`*_storage.backend`](https://github.com/reanahub/reana/tree/master/helm/reana) Helm value to `hostpath` and [`*_storage.hostpath.root_path`](https://github.com/reanahub/reana/tree/master/helm/reana) to the path of the directory you would like to use:
+
+```{ .yaml .copy-to-clipboard }
+shared_storage: # or `infrastructure_storage`
+  backend: hostpath
+  hostpath:
+    root_path: "/var/my-reana-directory"
+```
+
+## Migration to new infrastructure volume
+
+!!! danger
+    This procedure is delicate and it can lead to data loss. Carefully read the following instructions and, if possible, try them out on a test instance of REANA to make sure that everything works.
+
+!!! note
+    This guide requires to upgrade REANA a few times by using the `helm` tool. To make sure that you are running the correct commands, you can use the [`helm-diff`](https://github.com/databus23/helm-diff) plugin to inspect the changes that will be made to REANA's deployment.
+
+1. Make sure REANA is already updated to version 0.9.0 or greater. In the following steps, we will refer to this version as `$VERSION`.
+2. Disable the scheduling of new workflows by setting `REANA_MAX_CONCURRENT_BATCH_WORKFLOWS` to zero.
+
+    ```{ .console .copy-to-clipboard}
+    $ helm upgrade reana reanahub/reana --version $VERSION \
```

Comment:
Perhaps put the `--version` part onto a new line, so that we have one-line-one-value philosophy?